catroid/src/main/java/org/catrobat/catroid/stage/PreStageActivity.java

```diff
@@ -563,6 +595,31 @@ private void nfcInitialize() {
 		resourceInitialized();
 	}
 
+	private static ArrayList<Brick> getBricksRequieringResource(int resource) {
+		ArrayList<Brick> brickList = new ArrayList<Brick>();
+		ArrayList<Sprite> spriteList = (ArrayList<Sprite>) ProjectManager.getInstance().getCurrentProject()
+				.getSpriteListWithClones();
```

Comment:
The current implementation of [Project.getRequiredResources](https://github.com/Catrobat/Catroid/blob/8203e3adabc5583264b30b1f822282befee03092/catroid/src/main/java/org/catrobat/catroid/content/Project.java#L281-L299) is rubbish. It unnecessarily handles the ActionFactory and that's the reason why it's implemented that way. I'm not sure if I already wrote a ticket for that but I definitely have it on my private Catroid Cache list.

We definitely have to take a look at your implementation later (since I'm short on time) but currently it's just split up.